sklearn/linear_model/ridge.py

```diff
@@ -141,7 +145,17 @@ def _solve_cholesky_kernel(K, y, alpha, sample_weight=None):
         # Only one penalty, we can solve multi-target problems in one time.
         K.flat[::n_samples + 1] += alpha[0]
 
-        dual_coef = linalg.solve(K, y, sym_pos=True, overwrite_a=True)
+        try:
+            # Note: we must use overwrite_a=False in order to be able to
+            #       use the fall-back solution below in case a LinAlgError
+            #       is raised
+            dual_coef = linalg.solve(K, y, sym_pos=True,
+                                     overwrite_a=False)
+        except np.linalg.LinAlgError:
+            warnings.warn("Singular matrix in solving dual problem. Using "
+                          "pseudo inverse instead.")
+            Kinv = linalg.pinv2(K)
+            dual_coef = np.dot(Kinv, y)
```

Comment:
Re-adding @eickenberg's comment that got once again outdated:
I am sorry, I am unable to find the comments in the diff when this was still in the kernel ridge function. So my comment may have been handled, but I'd prefer to make sure: How small does alpha have to be and for what examples of K does this situation crop up? Wouldn't using a least squares solver (scipy.linalg.lstsq) in analogy to the try block be a semantically better choice? (again, really sorry if this has already been addressed/dismissed)